openhands/core/main.py

```diff
@@ -166,8 +168,12 @@ async def run_controller(
             EventSource.USER,
         )
     elif initial_state is None:
-        # init with the provided task
-        event_stream.add_event(MessageAction(content=task_str), EventSource.USER)
+        # init with the provided actions
+        for user_action in initial_user_actions:
+            assert isinstance(
+                user_action, Action
+            ), f'each element of initial_user_actions must be Action, got {type(user_action)}'
+            event_stream.add_event(user_action, EventSource.USER)
```

Comment:
In addition to @enyst comments, is this here being tested with actually more than 1 entry to exclude issues with LLM's?